File: src/main/java/com/enesgunumdogdu/hospital_management/service/DoctorService.java
package com.enesgunumdogdu.hospital_management.service;

import com.enesgunumdogdu.hospital_management.domain.dto.DoctorDto;

import java.util.List;

public interface DoctorService {
    
    List<DoctorDto> getAllDoctors();
    
    DoctorDto getDoctorById(Long id);
    
    DoctorDto createDoctor(DoctorDto doctorDto);
    
    DoctorDto updateDoctor(Long id, DoctorDto doctorDto);
    
    void deleteDoctor(Long id);
}

File: frontend-hospital-management/src/services/doctorService.js
import api from './api'

const doctorService = {
  getAllDoctors() {
    return api.get('/doctors')
  },

  getDoctorById(id) {
    return api.get(`/doctors/${id}`)
  },

  createDoctor(doctor) {
    return api.post('/doctors', doctor)
  },

  updateDoctor(id, doctor) {
    return api.put(`/doctors/${id}`, doctor)
  },

  deleteDoctor(id) {
    return api.delete(`/doctors/${id}`)
  }
}

export default doctorService